lib/compare.js

```diff
@@ -24,6 +24,15 @@ const equalObject = (obj1, obj2) => {
   return true;
 };
 
+const equalFunction = (func1, func2) => {
+  if (func1 === null || func2 === null) return func1 === func2;
+  const f1Str = func1.toString();
+  const f2Str = func2.toString();
+  return !f1Str.includes('[native code]') &&
```

Comment:
To make sure there will be less false negative cases, I'd also replace `String.prototype.includes()` here with a regular expression similar to this:
```js
/^function \w*\(\) \{ \[native code\] \}$/
```

Reply:
But this will introduce the possibility of having false-positive when two native functions have the same name?
Perhaps it's indeed worth having separate strict (this) and non-strict (with name equality check for native funtions) variants?